libraries/networking/src/udt/NetworkSocket.h

```diff
@@ -0,0 +1,163 @@
+//
+//  NetworkSocket.h
+//  libraries/networking/src/udt
+//
+//  Created by David Rowe on 21 Jun 2021.
+//  Copyright 2021 Vircadia contributors.
+//
+
+#ifndef vircadia_NetworkSocket_h
+#define vircadia_NetworkSocket_h
+
+#include <QObject>
+#include <QUdpSocket>
+
+#include <shared/WebRTC.h>
+
+#include "../HifiSockAddr.h"
+#include "../NodeType.h"
+#include "../SocketType.h"
+#if defined(WEBRTC_DATA_CHANNELS)
+#include "../webrtc/WebRTCSocket.h"
+#endif
+
+/// @addtogroup Networking
+/// @{
+
+
+/// @brief Multiplexes a QUdpSocket and a WebRTCSocket so that they appear as a single QUdpSocket-style socket.
+class NetworkSocket : public QObject {
```

Comment:
perhaps this should inherit from QAbstractSocket instead?

Reply:
`QAbstractSocket` inherits from `QIODevice` and no code from either of these is needed. It's not clear to me whether inheriting from `QAbstractSocket` would be beneficial or just overhead and a potential source of problems.